client/src/main/java/io/pravega/client/stream/EventStreamReader.java

```diff
@@ -25,6 +25,9 @@
      * timeout waiting for them to arrive. If none do, an EventRead will be returned with null for
      * {@link EventRead#getEvent()}. (As well as for most other fields)
      *
+     * An EventRead with null for {@link EventRead#getEvent()} is returned when the Reader has read all events up to the
```

Comment:
This is also null in a bunch of other cases. This API is basically impossible to code to.

Reply:
As discussed, null will be returned in such a scenario.

Reply:
Should we be throwing instead of returning null? Returning null for `EventRead#getEvent()` does not make the behavior very explicit and makes it prone to errors. Of course, we can blame the developer of the application for not reading the javadocs carefully, but I think it would be nicer on our end to throw and make the behavior explicit.

Reply:
Will fix this as part as part of the enhancement https://github.com/pravega/pravega/issues/2440